app/src/androidTest/java/ch/epfl/polychef/fragments/RateRecipeFragmentsEntryTest.java

```diff
@@ -0,0 +1,73 @@
+package ch.epfl.polychef.fragments;
+
+import android.content.Intent;
+
+import androidx.test.espresso.contrib.RecyclerViewActions;
+import androidx.test.espresso.matcher.RootMatchers;
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+import androidx.test.rule.ActivityTestRule;
+import androidx.test.runner.intercepting.SingleActivityFactory;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import ch.epfl.polychef.R;
+import ch.epfl.polychef.pages.EntryPage;
+import ch.epfl.polychef.pages.EntryPageTest;
+import ch.epfl.polychef.recipe.Recipe;
+import ch.epfl.polychef.recipe.RecipeTest;
+
+import static androidx.test.espresso.Espresso.onView;
+import static androidx.test.espresso.action.ViewActions.click;
+import static androidx.test.espresso.action.ViewActions.scrollTo;
+import static androidx.test.espresso.assertion.ViewAssertions.matches;
+import static androidx.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static androidx.test.espresso.matcher.ViewMatchers.withId;
+import static androidx.test.espresso.matcher.ViewMatchers.withText;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
+
+
+@RunWith(AndroidJUnit4.class)
+public class RateRecipeFragmentsEntryTest {
+
+    private SingleActivityFactory<EntryPage> fakeEntryPage = new SingleActivityFactory<EntryPage>(
+            EntryPage.class) {
+        @Override
+        protected EntryPage create(Intent intent) {
+            EntryPage activity = new EntryPageTest.FakeEntryPage();
+            return activity;
+        }
+    };
+
+    @Rule
+    public ActivityTestRule<EntryPage> intentsTestRuleEntry = new ActivityTestRule<>(fakeEntryPage, false,
+            true);
+
+    @Test
+    public void createInstanceOfRateRecipeFragmentDoesNotThrowError(){
+        RateRecipeFragment rrf=new RateRecipeFragment();
+    }
+
+    @Test
+    public void rateButtonIsDisplayedAndDisplayCorrectText(){
+
+        onView(withId(R.id.miniaturesOfflineList)).perform(RecyclerViewActions.actionOnItemAtPosition(0, click()));
+        onView(withId(R.id.buttonRate)).perform(scrollTo(),click());
+        onView(withId(R.id.buttonRate)).check(matches(isDisplayed()));
+        onView(withId(R.id.buttonRate)).check(matches(withText(R.string.RateButton)));
```

Comment:
It may be better for later to just remove the possibility to rate on the entry page, we can just do it like we did for the author and add:
```java
buttonRate.setVisibility(GONE);
```
when we are on the entryPage
(Same for the stars)

Reply:
Yes we can do this for later, or we can leave it here to tease the user and make him want to create a polychef account. We can discuss this at the next meeting

Reply:
Good idea, yes, we can discuss it later